app/queues/common/expiration-request.server.ts

```diff
@@ -0,0 +1,105 @@
+import { Worker, Queue, UnrecoverableError } from 'bullmq';
+
+import { redis } from '~/lib/redis.server';
+import logger from '~/lib/logger.server';
+import { prisma } from '~/db.server';
+
+import { addNotification } from '../notifications/notifications.server';
+import { deleteDnsRequest } from '../dns/dns-flow.server';
+import { RecordType as RecType } from '../notifications/expiration-notification.server';
+
+import type { ExpirationStatusPayload as ExpirationRequestPayload } from '../notifications/expiration-notification.server';
+
+const { EXPIRATION_JOB_REPEAT_FREQUENCY, JOB_REMOVAL_FREQUENCY } = process.env;
+declare global {
+  var __expiration_request_init__: boolean;
+}
+// queue name
+const expirationRequestQueueName = 'expiration-request';
+
+// query to fetch expired records from database based on type
+const getExpiredRecords = (type: RecType) => {
+  switch (type) {
+    case RecType.DnsRecord:
+      return prisma.record.findMany({
+        where: {
+          expiresAt: {
+            lt: new Date(),
+          },
+        },
+        select: {
+          id: true,
+          username: true,
+          type: true,
+          name: true,
+          value: true,
+          user: true,
+        },
+      });
+    case RecType.Certificate:
+      logger.info(`TODO: return expired certificates from the db`);
+      break;
+    default:
+      throw new Error(`Invalid recordType: ${type}`);
+  }
+};
+
+// queue initialization
+const expirationRequestQueue = new Queue<ExpirationRequestPayload>(expirationRequestQueueName, {
+  connection: redis,
+});
+
+export function addExpirationRequest(type: RecType) {
+  const jobName = `expirationRequestQueueName-${type}`;
+  return expirationRequestQueue.add(
+    jobName,
+    { type },
+    {
+      repeat: { every: Number(EXPIRATION_JOB_REPEAT_FREQUENCY) },
+      removeOnComplete: { age: Number(JOB_REMOVAL_FREQUENCY) },
+      removeOnFail: { age: Number(JOB_REMOVAL_FREQUENCY) },
+    }
+  );
+}
+
+// worker definition
+const expirationRequestWorker = new Worker<ExpirationRequestPayload>(
+  expirationRequestQueueName,
+  async (job) => {
+    const { type } = job.data;
+    switch (type) {
+      case RecType.DnsRecord:
+        try {
+          logger.info(`process DNS record expiration`);
+          let dnsRecords = await getExpiredRecords(type);
+          if (dnsRecords?.length) {
```

Comment:
This is unnecessary, you can `await Promise.all([])` and it will be ok.

Reply:
I'm confused why this is still here?

Reply:
> I'm confused why this is still here?

I don't know what you are referring to here.

Reply:
```ts
if (dnsRecords?.length) {
```

Reply:
Only try deleting if we get results back from a query, or is that just unnecessary since the job will fail?